### PR TITLE
Daml-LF: make MUL_NUMERIC and DIV_NUMERIC multi-scale

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
This PR advances the Numeric implementation #2289.

In this PR we change the signature of numeric builtins for multiplication and division as follows:

```
MUL_NUMERIC: ∀(s1 s2 s:nat). Numeric s1 → Numeric s2 → Numeric s
DIV_NUMERIC: ∀(s1 s2 s:nat). Numeric s1 → Numeric s2 → Numeric s
```

### Pull Request Checklist

- [X] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [X] Include appropriate tests
- [X] Set a descriptive title and thorough description
- [X] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [X] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate
- [X] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion/Primitives.hs [hunk 1]
```diff
@@ -164,9 +164,9 @@ convertPrim _ "BEAddDecimal" (TNumeric10 :-> TNumeric10 :-> TNumeric10) =
 convertPrim _ "BESubDecimal" (TNumeric10 :-> TNumeric10 :-> TNumeric10) =
     ETyApp (EBuiltin BESubNumeric) (TNat 10)
 convertPrim _ "BEMulDecimal" (TNumeric10 :-> TNumeric10 :-> TNumeric10) =
-    ETyApp (EBuiltin BEMulNumeric) (TNat 10)
+    ETyApp (ETyApp (ETyApp (EBuiltin BEMulNumeric) (TNat 10)) (TNat 10)) (TNat 10)
 convertPrim _ "BEDivDecimal" (TNumeric10 :-> TNumeric10 :-> TNumeric10) =
-    ETyApp (EBuiltin BEDivNumeric) (TNat 10)
+    ETyApp (ETyApp (ETyApp (EBuiltin BEMulNumeric) (TNat 10)) (TNat 10)) (TNat 10)
```
Comment (associahedron) [inline compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion/Primitives.hs]:
```suggestion
    ETyApp (ETyApp (ETyApp (EBuiltin BEDivNumeric) (TNat 10)) (TNat 10)) (TNat 10)
```

Comment (remyhaemmerle-da) [reply compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion/Primitives.hs]: I fixed before your commented it. Thanks for the catch anyway.

File: unreleased.rst [hunk 2]
```diff
@@ -12,3 +12,4 @@ HEAD — ongoing
 * [DAML Docs] suppress instance documentation when `--data-only` mode is requested
 + [DAML-LF] add CAST_NUMERIC and SHIFT_NUMERIC in DAML-LF 1.dev
 + [Sandbox] Dramatically increased performance of the ActiveContractService by only loading the contracts that the parties in the transaction filter are allowed to see.
++ [DAML-LF] change signature of MUL_NUMERIC and DIV_NUMERIC
```
Comment (hurryabit) [inline unreleased.rst]: I'm not sure we should track changes like this in the _user facing_ changelog. The numeric type is still very much under development and adding changelog entries for this adds a lot of noise and might give others the impression we're doing tons of breaking changes.
What do you think @gerolf-da?

Comment (gerolf-da) [reply unreleased.rst]: @hurryabit: sounds reasonable. Maybe for daml-lf 1.dev it makes sense to keep an internal changelog? this way we keep track of all the changes without adding it to the SDK changelog.

Comment (remyhaemmerle-da) [reply unreleased.rst]: @hurryabit You'r right. 
@georg-da We have already a change log for dev in the DAML-LF specification. It should be enough to track change here.

Comment (hurryabit) [reply unreleased.rst]: I propose that we promote the changes to DAML-LF 1.dev from the DAML-LF spec changelog to the proper changelog when we freeze a version of DAML-LF.

**Pull request merged**